### PR TITLE
add Damager

### PR DESCRIPTION
**Pull request by r0yzer**

### CODE REVIEW

**Comment by Hotkeyyy** [comment]
nein

File: src/main/resources/plugin.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@
     },
     "admingui": {
       "description": "Öffnet das Admin GUI."
+    },
+    "damage": {
+        description: "Stellt deinen Damager schaden ein (in halbe Herzen)"
```
Comment (jakobkmar) [inline src/main/resources/plugin.yml]: halben*

File: src/main/kotlin/net/axay/hglaborlobby/damager/DamagerDifficulty.kt [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        }
+        set(value) {
+            if (value) inconsistencyPlayers.add(this.name)
+            else inconsistencyPlayers.remove(this.name)
+        }
+
+    private var Player.inconsistencyRange: Pair<Int, Int>
+        get() {
+            if (this.name !in inconsistencyRanges) inconsistencyRanges[this.name] = 4 to 7
+            return inconsistencyRanges[this.name]!!
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/hglaborlobby/damager/DamagerDifficulty.kt]: don't use !!

File: src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        else -> false
+    }
+
+
+    private fun giveItems(player: Player) {
+        player.health = 20.0
+        player.feed()
+        player.saturation = 0F
+        player.inventory.clear()
+        player.gameMode = GameMode.ADVENTURE
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt]: remove gamemode change

File: src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        listen<EntityDamageEvent> {
+            if (it.entity !is Player) return@listen
+            val p = it.entity as Player
+            if (!p.isInDamager) return@listen
+
+            if (p.health - it.damage <= 0) {
+                if (p.soupsEaten >= 90) {
+                    p.playSound(p.location, Sound.ENTITY_PLAYER_LEVELUP, 1.0F, 1.0F)
+                    p.sendMessage("${KColors.GRAY}Du hast den Damager${KColors.GREEN} geschafft")
+                    if (playerDamage[p.name]!! >= 8.9 && !p.inconsistencyEnabled) broadcast("${KColors.GOLD}${p.name} §6hat den legendary Damager geschafft!")
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt]: don't use legacy color codes (like §6)

File: src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                    giveItems(p)
+                    p.soupsEaten = 0
+                } else {
+                    p.inventory.clear()
+                    p.teleport(Location(Bukkit.getWorld("world"), -495.5, 63.0, 452.5)) // TODO add correct coordinates
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt]: teleport to damager warp

File: src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+                    p.heal()
+                    p.sendMessage("${KColors.GRAY}Du hast den Damager ${KColors.RED}nicht${KColors.GRAY} geschafft")
+                    p.playSound(p.location, Sound.ENTITY_OCELOT_HURT, 1.0F, 1.0F)
+                    p.soupsEaten = 0
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt]: remove player from list (also on leave)

File: src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+                if (damager.isInArea(player.location)) {
+                    if (player.name !in playersInDamager) {
+                        sync { giveItems(player) }
+                        playersInDamager.plusAssign(player.name)
+                        if (player.name !in playerDamage.keys) playerDamage[player.name] = 5.0 // add player to playerDamage Map if he was not in the damager before
+                    }
+                    break
+                } else {
+                    if (player.name in playersInDamager && !player.isInDamager) {
+                        sync { player.gameMode = GameMode.SURVIVAL }
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/hglaborlobby/damager/Damager.kt]: remove gamemode change

File: src/main/kotlin/net/axay/hglaborlobby/damager/DamagerDifficulty.kt [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+            range.first <= 0 ||
+            range.second >= 20
+        )
+            player.inconsistencyRange = oldRange
+    }
+
+    fun enable() {
+        listen<PlayerInteractEvent> {
+            if (it.clickedBlock?.type != Material.OAK_WALL_SIGN) return@listen
+            val sign = it.clickedBlock!!.state as Sign
```
Comment (jakobkmar) [inline src/main/kotlin/net/axay/hglaborlobby/damager/DamagerDifficulty.kt]: don't use !!

**Review by jakobkmar — CHANGES_REQUESTED** [review]
A damager is a very cool feature.
Please address the requested changes 🙂

**Review by jakobkmar — APPROVED** [review]
Let's just hope the best 😃

**Pull request merged**